client/components/tinymce/plugins/wpcom-help/help-modal.jsx

```diff
@@ -26,46 +26,114 @@ class HelpModal extends React.Component {
-			{ r: this.props.translate( 'Align right' ), j: this.props.translate( 'Justify' ) },
-			{ d: this.props.translate( 'Strikethrough' ), q: this.props.translate( 'Blockquote' ) },
-			{ u: this.props.translate( 'Bullet list' ), o: this.props.translate( 'Numbered list' ) },
-			{ a: this.props.translate( 'Insert/edit link' ), s: this.props.translate( 'Remove link' ) },
-			{
-				m: this.props.translate( 'Insert/edit image' ),
-				t: this.props.translate( 'Insert Read More tag' ),
-			},
-			{ h: this.props.translate( 'Keyboard Shortcuts' ), x: this.props.translate( 'Code' ) },
-			{ p: this.props.translate( 'Insert Page Break tag' ) },
+			[
+				{ key: 1, action: translate( 'Heading 1' ) },
+				{ key: 2, action: translate( 'Heading 2' ) },
+			],
+			[
+				{ key: 3, action: translate( 'Heading 3' ) },
+				{ key: 4, action: translate( 'Heading 4' ) },
+			],
+			[
+				{ key: 5, action: translate( 'Heading 5' ) },
+				{ key: 6, action: translate( 'Heading 6' ) },
+			],
+			[
+				{ key: 'l', action: translate( 'Align left' ) },
+				{ key: 'c', action: translate( 'Align center' ) },
+			],
+			[
+				{ key: 'r', action: translate( 'Align right' ) },
+				{ key: 'j', action: translate( 'Justify' ) },
+			],
+			[
+				{ key: 'd', action: translate( 'Strikethrough' ) },
+				{ key: 'q', action: translate( 'Blockquote' ) },
+			],
+			[
+				{ key: 'u', action: translate( 'Bulleted list' ) },
+				{ key: 'o', action: translate( 'Numbered list' ) },
+			],
+			[
+				{ key: 'a', action: translate( 'Insert/edit link' ) },
+				{ key: 's', action: translate( 'Remove link' ) },
+			],
+			[
+				{ key: 'm', action: translate( 'Insert/edit image' ) },
+				{ key: 't', action: translate( 'Insert Read More tag' ) },
+			],
+			[
+				{ key: 'h', action: translate( 'Keyboard Shortcuts' ) },
+				{ key: 'x', action: translate( 'Code' ) },
+			],
+			[ { key: 'p', action: translate( 'Insert Page Break tag' ) } ],
+		];
+	};
+
+	spaceFormatShortcuts = () => {
+		const { translate } = this.props;
+		return [
+			[
+				{ key: '*', action: translate( 'Bulleted list' ) },
+				{ key: '1.', action: translate( 'Numbered list' ) },
+			],
+			[
+				{ key: '-', action: translate( 'Bulleted list' ) },
+				{ key: '1)', action: translate( 'Numbered list' ) },
+			],
+		];
+	};
+
+	enterFormatShortcuts = () => {
+		const { translate } = this.props;
+		return [
+			[
+				{ key: '>', action: translate( 'Blockquote' ) },
+				{ key: '##', action: translate( 'Heading 2' ) },
+			],
+			[
+				{ key: '###', action: translate( 'Heading 3' ) },
+				{ key: '####', action: translate( 'Heading 4' ) },
+			],
+			[
+				{ key: '#####', action: translate( 'Heading 5' ) },
+				{ key: '######', action: translate( 'Heading 6' ) },
+			],
+			[ { key: '---', action: translate( 'Horizontal line' ) } ],
 		];
 	};
 
 	renderRow = ( row, index ) => {
 		let columns = [];
 
-		forEach( row, ( text, key ) => {
+		forEach( row, ( obj, i ) => {
```

Comment:
`obj` and `i` are ambiguous here, especially given `i` is an abbreviation of `index` which we already have in the parenting scope. I find that scenarios like this act as a clue that something could be named differently to better show intent.

I like the move to change the data structure to be an array of arrays. I think that represents the data in a way that is more true to form. The outer array represents the 'rows' and the inner array represents the 'cells' that make up the columns implicitly. Thinking of it that way, I think `cell` or `rowCell` or something along those lines might be a better name for the iteratee and then a derivative of this could be made for it's index (`cellIndex`). For the sake of extra clarity after introducing a second index I'd be tempted to also change the current `index` to `rowIndex`.

An alternative to fiddling with the index variables would be to just use `cell.key` to build the list-item `key` prop (slightly confusing!).